television/utils/image.rs

```diff
@@ -0,0 +1,255 @@
+use image::imageops::FilterType;
+use image::{DynamicImage, GenericImageView, Pixel, Rgba};
+use ratatui::buffer::{Buffer, Cell};
+use ratatui::layout::{Position, Rect};
+use ratatui::prelude::Color;
+use ratatui::widgets::{Block, Widget};
+use std::fmt::Debug;
+use std::hash::{Hash, Hasher};
+use std::sync::{Arc, Mutex};
+
+static PIXEL_STRING: &str = "▀";
+const FILTER_TYPE: FilterType = FilterType::Lanczos3;
+
+// use to reduce the size of the image before storing it
+const CACHED_WIDTH: u32 = 128;
+const CACHED_HEIGHT: u32 = 128;
+
+const GRAY: Rgba<u8> = Rgba([242, 242, 242, 255]);
+const WHITE: Rgba<u8> = Rgba([255, 255, 255, 255]);
+
+pub struct ImagePreviewWidget<'a> {
+    block: Block<'a>,
+    displayed_image: Arc<Mutex<Option<DisplayedImage>>>,
+}
+impl<'a> ImagePreviewWidget<'a> {
+    pub fn new(
+        displayed_image: Arc<Mutex<Option<DisplayedImage>>>,
+        block: ratatui::widgets::Block<'a>,
+    ) -> ImagePreviewWidget<'a> {
+        ImagePreviewWidget {
+            block,
+            displayed_image,
+        }
+    }
+}
+impl Widget for ImagePreviewWidget<'_> {
+    fn render(self, area: Rect, buf: &mut Buffer)
+    where
+        Self: Sized,
+    {
+        // I also put the render of the block to be similar has the preview paragraph, but I believe they are both useless since the block is already rendered
+        let inner = self.block.inner(area);
+        self.block.render(area, buf);
```

Comment:
```suggestion
        let inner = self.block.inner(area);
```

this does look unnecessary indeed